code/cheats_table/cheats_table.cpp

```diff
@@ -0,0 +1,110 @@
+#include "cheats_table.h"
+#include "parse/parselo.h"
+
+void cheat_table_init() {
+	customCheats.clear();
+
+	// Add default Spawn Ship Cheat (Volition Bravos).
+	// Won't interfere with Total Conversions, since if the ship doesn't exist, the game will just ignore the cheat.
+	// The other default cheats do not use the Custom Cheat Framework and are still handled in keycontrol.cpp
+	std::unique_ptr<CustomCheat> bravos(new SpawnShipCheat("arrrrwalktheplank", "Walk the plank", false, "Volition Bravos", "Volition Bravos"));
+	customCheats.emplace(SCP_string(bravos->cheatCode), std::move(bravos));
+
+	if (cf_exists_full("cheats.tbl", CF_TYPE_TABLES)) {
+		parse_cheat_table("cheats.tbl");
+	}
+	parse_modular_table("*-cht.tbm", parse_cheat_table);
+}
+
+void parse_cheat_table(const char* filename) {
+	try {
+		read_file_text(filename, CF_TYPE_TABLES);
+		reset_parse();
+		optional_string("#CUSTOM CHEATS");
+
+		while (optional_string("$Cheat:")) {
+			SCP_string code, msg, shipClass, shipName;
+			bool requireCheats = false;
+			bool shipSpawn = false;
+			required_string("+Code:");
+			stuff_string(code, F_RAW);
+
+			//We are limited to the buffer size used for the traditional cheats. We can still use short cheats, though.
+			if (code.length() > CHEAT_BUFFER_LEN) {
+				Warning(LOCATION, "Cheat code %s is too long. It will be cut off to maximum length (%i characters).", code.c_str(), CHEAT_BUFFER_LEN);
+				code = code.substr(0, CHEAT_BUFFER_LEN);
+			}
+
+			required_string("+Message:");
+			stuff_string(msg, F_RAW);
```

Comment:
Read an ``F_MULTITEXT`` here, maybe?

Reply:
This causes the default parsing to fail (seems that it needs a specific terminator in the table). On the other hand, there's an `F_MESSAGE`, which works fine

Reply:
``F_MESSAGE`` is obsolete though. I wonder what the proper course of action is here.

Reply:
It's obsolete for in-mission messages.  It's still used for a few other things though.  I'd be ok with using it.

Reply:
You could also use F_NAME or F_LNAME, depending on whether you wanted localization.